modules/spechla/1.0/spechla.smk

```diff
@@ -0,0 +1,310 @@
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["spechla"]`
+CFG = op.setup_module(
+    name = "spechla",
+    version = "1.0",
+    subdirectories = ["inputs", "hla_reads", "spechla", "loh", "outputs"],
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _spechla_input_bam,
+    _spechla_setup,
+    _spechla_index,
+    _spechla_output_txt,
+    _spechla_all,
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _spechla_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bai",
+        crai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.crai"
+    group: 
+        "symlink"
+    run:
+        op.absolute_symlink(input.bam, output.bam)
+        op.absolute_symlink(input.bai, output.bai)
+        op.absolute_symlink(input.bai, output.crai)
+
+
+# Example variant calling rule (multi-threaded; must be run on compute server/cluster)
+rule _spechla_setup:
+    output:
+        txt = touch(CFG["dirs"]["inputs"] + "SpecHLA/clone.done")
+    log:
+        stdout = CFG["logs"]["inputs"] + "git_clone.stdout.log"
+    params:
+        scriptdir = CFG["dirs"]["inputs"] + "SpecHLA"
+    shell:
+        op.as_one_line("""
+        git clone git@github.com:deepomicslab/SpecHLA.git --depth 1  {params.scriptdir} > {log.stdout} 2>&1 && 
+        chmod +x -R {params.scriptdir}/bin/*
+        """)
+
+rule _spechla_index:
+    input:
+        txt = CFG["dirs"]["inputs"] + "SpecHLA/clone.done"
+    output:
+        txt = touch(CFG["dirs"]["inputs"] + "SpecHLA/index.done")
+    log:
+        stdout = CFG["logs"]["inputs"] + "index.stdout.log"
+    params:
+        scriptdir = CFG["dirs"]["inputs"] + "SpecHLA"
+    threads:
+        CFG["threads"]["setup"]
+    resources:
+        **CFG["resources"]["setup"]  
+    shell:
+        op.as_one_line("""
+        unset LD_LIBRARY_PATH && unset LIBRARY_PATH &&
+        bash {params.scriptdir}/index.sh > {log.stdout} 2>&1
+        """)
+
+# Extract HLA reads from bam and cram
+def _which_genome(wildcards):
+    if "38" in str({wildcards.genome_build}):
+        return "hg38"
+    else:
+        return "hg19"
+
+rule _spechla_extract_reads:
+    input:
+        txt = CFG["dirs"]["inputs"] + "SpecHLA/index.done",
```

Comment:
Here (and in the rules below)  the inputs also need a switch to the str(rule..output....) so it is consistent with the rest of the modules